collaborative-plonk/src/types.rs

```diff
@@ -0,0 +1,137 @@
+use ark_ec::AffineRepr;
+use std::marker::PhantomData;
+
+use ark_ec::pairing::Pairing;
+use ark_ff::PrimeField;
+use ark_serialize::CanonicalSerialize;
+use mpc_core::traits::PrimeFieldMpcProtocol;
+use sha3::{Digest, Keccak256};
+
+use crate::FieldShareVec;
+
+pub(crate) type Keccak256Transcript<P> = Transcript<Keccak256, P>;
+pub(crate) struct Transcript<D, P>
+where
+    D: Digest,
+    P: Pairing,
+{
+    digest: D,
+    phantom_data: PhantomData<P>,
+}
+
+impl<P: Pairing> Default for Keccak256Transcript<P> {
+    fn default() -> Self {
+        Self {
+            digest: Default::default(),
+            phantom_data: Default::default(),
+        }
+    }
+}
+
+impl<D, P> Transcript<D, P>
+where
+    D: Digest,
+    P: Pairing,
+{
+    pub(crate) fn add_scalar(&mut self, scalar: P::ScalarField) {
+        let mut buf = vec![];
+        scalar
+            //.lift_montgomery() Check if we need this or not. For round2 we do not need it
+            .serialize_uncompressed(&mut buf)
+            .expect("Can Fr write into Vec<u8>");
+        buf.reverse();
+        self.digest.update(&buf);
+    }
+
+    pub(crate) fn add_point(&mut self, point: P::G1Affine) {
+        let bits: usize = P::BaseField::MODULUS_BIT_SIZE
+            .try_into()
+            .expect("u32 fits into usize");
+        let mut buf = Vec::with_capacity(bits);
+        if let Some((x, y)) = point.xy() {
+            x.serialize_uncompressed(&mut buf)
+                .expect("Can Fq write into Vec<u8>");
+            buf.reverse();
+            self.digest.update(&buf);
+            buf.clear();
+            y.serialize_uncompressed(&mut buf)
+                .expect("Can Fq write into Vec<u8>");
+            buf.reverse();
+            self.digest.update(&buf);
+        } else {
+            // we are at infinity
```

Comment:
Could use a comment on what the circom repr for infinity is compared to the arkworks one.

Reply:
added a comment